examples/gnrc_tcp_cli/main.c

```diff
@@ -0,0 +1,235 @@
+/*
+ * Copyright (C) 2015 Simon Brummer
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <errno.h>
+#include "net/af.h"
+#include "random.h"
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/tcp.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#ifndef TARGET_ADDR
+#define TARGET_ADDR "fe80::ccea:ebff:fe12:f132"
+#endif
+
+#ifndef TARGET_PORT
+#define TARGET_PORT 80 /* Target Port */
+#endif
+
+#ifndef CONNS
+#define CONNS 1        /* Number of parallel connections */
+#endif
+
+#ifndef NBYTE
+#define NBYTE (2048)   /* Amount of data to transmit */
+#endif
+
+#ifndef CYCLES
+#define CYCLES 10      /* Number of repeated Cycles per thread */
+#endif
+
+uint8_t bufs[CONNS][NBYTE];
+uint8_t stacks[CONNS][THREAD_STACKSIZE_DEFAULT + THREAD_EXTRA_STACKSIZE_PRINTF];
+
+void *cli_thread(void *arg);
+
+int main(void)
+{
+    printf("\nStarting Client Threads. Config is: TARGET_ADDR=%s, TARGET_PORT=%d, ", TARGET_ADDR, TARGET_PORT);
+    printf("CONNS=%d, NBYTE=%d, CYCLES=%d\n\n", CONNS, NBYTE, CYCLES );
+
+    /* Start Connection Handling Threads */
+    for (int i = 0; i < CONNS; i += 1) {
+        thread_create((char *)stacks[i], sizeof(stacks[i]), THREAD_PRIORITY_MAIN, 0, cli_thread,
+                      (void *)i, NULL);
+    }
+
+    return 0;
+}
+
+void *cli_thread(void *arg)
+{
+    /* Test Program variables */
+    int32_t tid = (int) arg;
+    uint32_t rcvd = 0;
+    uint32_t sent = 0;
+    uint32_t cycles = 0;
+    uint32_t cycles_ok = 0;
+
+    /* TCP specific variables */
+    ipv6_addr_t local_addr;
+    ipv6_addr_t peer_addr;
+    gnrc_tcp_tcb_t tcb;
+
+    /* Target Peer Address Information */
+    ipv6_addr_t target_addr;
+    uint16_t target_port;
+
+    /* Initialize Target Information */
+    ipv6_addr_from_str(&target_addr, TARGET_ADDR);
+    target_port = TARGET_PORT;
+
+    printf("client thread running: TID=%"PRIi32"\n", tid);
+    while (cycles < CYCLES) {
+        /* Initialize tcb struct */
+        int32_t ret = gnrc_tcp_tcb_init(&tcb, AF_INET6, (uint8_t *) &local_addr, (uint8_t *) &peer_addr);
+        switch (ret) {
+            case 0:
+                DEBUG("TID=%"PRIi32" : gnrc_tcp_tcb_init() : 0 : ok\n", tid);
+                break;
+
+            case -EAFNOSUPPORT:
+                printf("TID=%"PRIi32" : gnrc_tcp_tcb_init() : -EAFNOSUPPORT\n", tid);
+                continue;
+
+            default:
+                printf("TID=%"PRIi32" : gnrc_tcp_tcb_init() : %"PRIi32"\n", tid, ret);
+                continue;
+        }
+
+        /* Connect to Peer */
+        ret = gnrc_tcp_open_active(&tcb, AF_INET6, (uint8_t*) &target_addr, target_port, 0);
+        switch (ret) {
```

Comment:
same here, the program should wait before retry, just in case the target is not ready yet.